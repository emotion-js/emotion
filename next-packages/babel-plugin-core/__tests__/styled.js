import babelTester from 'babel-tester'
import plugin from '@emotion/babel-plugin-core'
import path from 'path'

babelTester(
  '@emotion/babel-plugin-core styled',
  path.join(__dirname, '..', '..', 'styled.macro', '__tests__'),
  {
    plugins: [plugin],
    transform: src => src.replace(/\.macro/g, '')
  }
)
