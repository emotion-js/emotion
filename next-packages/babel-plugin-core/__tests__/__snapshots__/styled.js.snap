// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@emotion/babel-plugin-core styled basic 1`] = `
"import styled from '@emotion/styled'

const SomeComponent = styled.div\`
  color: hotpink;
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

const SomeComponent = _styled(\\"div\\", {
  target: \\"eddbda30\\",
  label: \\"SomeComponent\\"
})(\\"color:hotpink;\\");"
`;

exports[`@emotion/babel-plugin-core styled call-expression 1`] = `
"import styled from '@emotion/styled'

const SomeComponent = styled('div')\`
  color: hotpink;
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

const SomeComponent = _styled('div', {
  target: \\"e1wwfyhh0\\",
  label: \\"SomeComponent\\"
})(\\"color:hotpink;\\");"
`;

exports[`@emotion/babel-plugin-core styled label 1`] = `
"import styled from '@emotion/styled'

class Thing {
  static SomeComponent = styled.div\`
    color: hotpink;
  \`
  BadIdeaComponent = styled.div\`
    background: hotpink;
  \`
}


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

class Thing {
  static SomeComponent = _styled(\\"div\\", {
    target: \\"erkd9g00\\",
    label: \\"SomeComponent\\"
  })(\\"color:hotpink;\\");
  BadIdeaComponent = _styled(\\"div\\", {
    target: \\"erkd9g01\\",
    label: \\"BadIdeaComponent\\"
  })(\\"background:hotpink;\\");
}"
`;

exports[`@emotion/babel-plugin-core styled multiple-calls 1`] = `
"import styled from '@emotion/styled'

const SomeComponent = styled.div\`
  color: hotpink;
\`

const SomeOtherComponent = styled.button\`
  color: green;
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

const SomeComponent = _styled(\\"div\\", {
  target: \\"e15zzbw00\\",
  label: \\"SomeComponent\\"
})(\\"color:hotpink;\\");

const SomeOtherComponent = _styled(\\"button\\", {
  target: \\"e15zzbw01\\",
  label: \\"SomeOtherComponent\\"
})(\\"color:green;\\");"
`;

exports[`@emotion/babel-plugin-core styled object 1`] = `
"import styled from '@emotion/styled'

const SomeComponent = styled.div({ color: 'hotpink' })


      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from \\"@emotion/styled-base\\";

const SomeComponent = _styled(\\"div\\", {
  target: \\"es0jczh0\\",
  label: \\"SomeComponent\\"
})({
  color: 'hotpink'
});"
`;
